backend/server/src/services/tournamentService.ts

```diff
@@ -88,6 +88,183 @@ export class TournamentService {
     return await newTournament.toObject();
   }
 
+  public async addTeamToTournament(
+    tournamentId: string,
+    teamName: string,
+    creatorId: string
+  ): Promise<Tournament> {
+    const tournament = await TournamentModel.findById(tournamentId);
+
+    if (tournament === null || tournament === undefined) {
+      throw new NotFoundError({
+        message: "No tournaments found"
+      });
+    }
+
+    const teamExists = (tournament.teams ?? []).some(
+      (team) => team.name === teamName
+    );
+    if (teamExists) {
+      throw new BadRequestError({
+        message: "A team with this name already exists in the tournament"
+      });
+    }
+
+    if (tournament.teams != null) {
```

Comment:
Here pretty much same but the opposite, `if (tournament.teams)`. Code works same so no need to do additional commits